sky/clouds/service_catalog/aws_catalog.py

```diff
@@ -35,17 +36,28 @@
                                 pull_frequency_hours=_PULL_FREQUENCY_HOURS)
 
 
-def _apply_az_mapping(df: 'pd.DataFrame') -> 'pd.DataFrame':
+def _apply_az_mapping(df: pd.DataFrame) -> pd.DataFrame:
     """Maps zone IDs (use1-az1) to zone names (us-east-1x).
 
-    Such mappings are account-specific and determined by AWS.
+    Such mappings are account-specific and determined by AWS. The mpaaing will
+    also remove the regions that are not supported by the user account. If the
+    user does not have AWS credentials configured, we use a default mapping.
 
     Returns:
         A dataframe with column 'AvailabilityZone' that's correctly replaced
         with the zone name (e.g. us-east-1a).
     """
     az_mapping_path = common.get_catalog_path('aws/az_mappings.csv')
     if not os.path.exists(az_mapping_path):
+        aws_enabled, _ = aws.AWS().check_credentials()
+        if not aws_enabled:
+            logger.debug(
+                f'{colorama.Style.DIM}Availability zone mapping is not '
+                f'available because AWS credentials are not configured. '
+                f'Using the default mapping.{colorama.Style.RESET_ALL}')
```

Comment:
There may be an edge case where `check_credentials()` returns False, but in fact `fetch_aws.fetch_availability_zone_mappings()` has the necessary credential files and go through? E.g., the `aws` CLI is not installed, but https://github.com/skypilot-org/skypilot/blob/d9fd2794c2db1befa529fd7e1b7e5aae7b23d1eb/sky/clouds/service_catalog/data_fetchers/fetch_aws.py#L106 may go through.

How about we try-catch L69?